cmd/atlasci/internal/ci/ci.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package ci
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"sort"
+
+	"ariga.io/atlas/sql/migrate"
+	"ariga.io/atlas/sql/schema"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/utils/merkletrie"
+)
+
+type (
+	// A ChangeDetector takes a migration directory and splits it into the "base" files (already merged) and new ones.
+	ChangeDetector interface {
+		// DetectChanges splits the files of a migration directory into the "base" files (already merged) and new ones.
+		DetectChanges(context.Context) ([]migrate.File, []migrate.File, error)
+	}
+	// A ChangeLoader takes a set of migration files and will create multiple schema.Changes out of it.
+	// It will also label migration files as either "generated" or "handcrafted".
+	ChangeLoader interface {
+		// LoadChanges converts each of the given migration files into one Changes.
+		LoadChanges(context.Context, []migrate.File) ([]Changes, error)
+	}
+	// DirScanner stitches migrate.Dir and migrate.Scanner into one interface.
+	DirScanner interface {
+		migrate.Dir
+		migrate.Scanner
+	}
+	// Change wraps a schema.Change and adds metadata to it.
+	Change struct {
+		schema.Change
+		// Generated determines if this file was solely generated by Atlas and not edited later on by a human.
+		Generated bool
+	}
+	// Changes is a list of Change.
+	Changes []Change
+)
+
+type (
+	// GitChangeDetector implements the ChangeDetector interface by utilizing a git repository.
+	GitChangeDetector struct {
+		root string     // path to the git repository root
+		base string     // name of the base branch (e.g. master)
+		path string     // path of the migration directory relative to the repository root (in slash notation)
+		dir  DirScanner // the migration directory to load migration files from
```

Comment:
```suggestion
		root string     // path to the git repository root.
		base string     // name of the base branch (e.g. master).
		path string     // path of the migration directory relative to the repository root (in slash notation).
		dir  DirScanner // the migration directory to load migration files from.
```

Reply:
I prefer to not add punctuation, since these comments are no sentences.